### PR TITLE
[Xamarin.Android.Build.Tasks] Add MSBuild support for DIM.

### PR DESCRIPTION
**Pull request by jpobst**
Adds MSBuild support for DIM to Android Bindings projects.

To enable, add the following to a binding project:
```
<LangVersion>preview</LangVersion>
<_EnableInterfaceMembers>True</_EnableInterfaceMembers>
```

### CODE REVIEW

File: tests/CodeGen-Binding/Xamarin.Android.DefaultInterfaceMethods-Tests/DimBindingTests.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+			Assert.AreEqual (2, iface.Bar);
+			Assert.DoesNotThrow (() => iface.Bar = 5);
+
+			Assert.AreEqual (0, iface.InvokeFoo ());
+
+			Assert.Throws<UnsupportedOperationException> (() => iface.ToImplement ());
+		}
+
+		[Test]
+		public void TestOverriddenDefaultInterfaceMethods ()
```
[the remark this answers is not in the document]
Comment (jpobst) [reply tests/CodeGen-Binding/Xamarin.Android.DefaultInterfaceMethods-Tests/DimBindingTests.cs]: Added, which revealed a new issue that required fixing. (https://github.com/xamarin/java.interop/pull/486)

**Comment by jpobst** [comment]
I'm not aware of any benefit to `Xamarin.Android.DefaultInterfaceMethods-Lib`. I merely ported it from https://github.com/xamarin/xamarin-android/pull/1939.  I suspect it was needed at the time because it required a custom compiler with DIM support.  Now that DIM is built into shipping compilers and runtimes that isn't needed.  Moved tests to `McwGen` and `JcwGen`.

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Bindings.targets [hunk 2]
```diff
@@ -494,6 +494,8 @@ Copyright (C) 2012 Xamarin Inc. All rights reserved.
       ToolPath="$(MonoAndroidToolsDirectory)"
       ToolExe="$(BindingsGeneratorToolExe)"
       UseShortFileNames="$(UseShortGeneratorFileNames)"
+      LangVersion="$(LangVersion)"
+      EnableInterfaceMembersPreview="$(EnableInterfaceMembersPreview)"
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Xamarin.Android.Bindings.targets]: Do we need to document this yet? https://github.com/xamarin/xamarin-android/blob/master/Documentation/guides/BuildProcess.md

Comment (dellis1972) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Bindings.targets]: Good catch, I forgot about documentation.

Comment (jonpryor) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Bindings.targets]: @jpobst: please update this section: https://github.com/xamarin/xamarin-android/blob/master/Documentation/guides/BuildProcess.md#binding-project-build-properties

To document the new `$(EnableInterfaceMembersPreview)` property.

Additionally, convention is that public properties begin with `Android`.

The `Preview` suffix is...disturbing?  What's the longevity for this property?

Should this instead be a `$(_EnableInterfaceMembers)` property?  A `_` prefix is "private"/"subject to change without notice" designator, so maybe that's what we should use?

Comment (jpobst) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Bindings.targets]: The intention is to explicitly denote that this field is temporary and not something we are going to support for the rest of time.  I am fine with a "private" property if that conveys the same idea.  :)

**Comment by jonpryor** [comment]
[`Xamarin.Android.Bcl_Tests-NUnit` crashed](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=3026697&view=ms.vss-test-web.build-test-results-tab&runId=8351948&paneView=debug), but the associated logical doesn't really mention a "crash":

```
09-06 22:32:49.779  3242  5090 I NUnit   : Passed: 22173, Failed: 0, Skipped: 419, Inconclusive: 1, Total: 0, Filtered: 0
09-06 22:32:49.788  1954  1973 I ActivityManager: Force stopping Xamarin.Android.Bcl_Tests appid=10103 user=0: finished inst
09-06 22:32:49.789  1954  1973 I ActivityManager: Killing 3242:Xamarin.Android.Bcl_Tests/u0a103 (adj 0): stop Xamarin.Android.Bcl_Tests
...
09-06 22:32:49.881  1651  1651 I Zygote  : Process 3242 exited due to signal 9 (Killed)
```

It was force-stopped, for some reason, which was detected as a crash.

Ignoring.

**Pull request merged**